src/Facemesh/index.js

```diff
@@ -0,0 +1,225 @@
+
+class Facemesh {
+  /**
+   * An options object to configure Facemesh settings
+   * @typedef {Object} configOptions
+   * @property {number} maxFacess - The maximum number of faces to detect. Defaults to 2.
+   * @property {boolean} refineLandmarks - Refine the ladmarks. Defaults to false.   
+   * @property {boolean} flipHorizontal - Flip the result horizontally. Defaults to false.
+   * @property {string} runtime - The runtime to use. "mediapipe"(default) or "tfjs".
+   *
+   * // For using custom or offline models
+   * @property {string} solutionPath - The file path or URL to the model.
+   */
+
+  /**
+   * Create Facemesh.
+   * @param {configOptions} options - An object with options.
+   * @param {function} callback - A callback to be called when the model is ready.
+   *
+   * @private
+   */
+  constructor(options, callback) {
+    // for compatibility with p5's preload()
+    if (this.p5PreLoadExists()) window._incrementPreload();
+
+    this.model = null;
+    this.config = options;
+    this.runtimeConfig = {};
+    this.detectMedia = null;
+    this.detectCallback = null;
+
+    // flags for detectStart() and detectStop()
+    this.detecting = false; // true when detection loop is running
+    this.signalStop = false; // true when detectStop() is called and detecting is true
+    this.prevCall = ""; // "start" or "stop", used for giving warning messages with detectStart() is called twice in a row
+
+    this.ready = callCallback(this.loadModel(), callback);
+  }
+
+  /**
+   * Load the model and set it to this.model
+   * @return {this} the Facemesh model.
+   *
+   * @private
+   */
+  async loadModel() {
+    const pipeline = faceLandmarksDetection.SupportedModels.MediaPipeFaceMesh;
+    //filter out model config options
+    const modelConfig = {
+        runtime: 'mediapipe',
+        solutionPath: 'https://cdn.jsdelivr.net/npm/@mediapipe/face_mesh',
+        maxFaces: this.config?.maxFaces ?? 1, 
+        refineLandmarks: this.config?.refineLandmarks ?? false,
+      };
+    this.runtimeConfig = {
+      flipHorizontal: this.config?.flipHorizontal ?? false,
+    };
+
+    await tf.ready();
+    this.model = await faceLandmarksDetection.createDetector(pipeline, modelConfig);
+
+    // for compatibility with p5's preload()
+    if (this.p5PreLoadExists) window._decrementPreload();
+
+    return this;
+  }
+
+  /**
+   * Asynchronously output a single face prediction result when called
+   * @param {*} [media] - An HMTL or p5.js image, video, or canvas element to run the prediction on.
+   * @param {function} [callback] - A callback function to handle the predictions.
+   * @returns {Promise<Array>} an array of predictions.
+   */
+  async detect(...inputs) {
+    //Parse out the input parameters
+    const argumentObject = handleArguments(...inputs);
+    argumentObject.require(
+      "image",
+      "An html or p5.js image, video, or canvas element argument is required for detectStart()."
```

Comment:
```suggestion
      "An html or p5.js image, video, or canvas element argument is required for detect()."
```

Just a small thing I noticed when working on handpose. This error message is supposed to say "detect()" since it is inside the `detect` function.